### PR TITLE
fix: OpenAPI specs for the global Plugins and correction in the PatchPlugin filtering

### PR DESCRIPTION
**Pull request by RajeevRanjan27**
fix - Added Open API specs for Global Plugins  related endpoints and Improvements in GlobalPlugin API Patch On Update Action Request #4789 

# Description

This pull request includes the Open API specifications for the endpoints related to the global plugins part. The addition of these specifications aims to improve documentation, making it easier for developers and users to understand how to interact with the API. This effort is part of an ongoing task to enhance our documentation for better user experience and developer engagement. No new dependencies are required for this change.

This also contains the fix of the plugin related filtering during the PatchPlugin Request to make sure free of any error.

Fixes https://github.com/devtron-labs/devtron/issues/4789
Fixes devtron-labs/devtron#4789

# Peer Review

## Checklist:

* [x] The title of the PR accurately describes the documentation changes.
* [x] This PR is primarily a documentation update. If additional code changes were made, they are minimal and necessary for the documentation effort.
* [x] I've updated the Open API specifications as required by this PR.
* [x] I have performed a self-review of the Open API documentation to ensure accuracy and completeness.
* [x] I have validated the Open API specs using relevant tools ( Swagger UI) to ensure they render correctly and are in compliance with the Open API standard.
* [x] The documentation has been reviewed by buddy (Prakash) to ensure its working  and description.
* [x] I have commented on sections of the specs that might be complex or require additional context for understanding.
* [x] I have ensured that the documentation is accessible and usable for all intended user roles, including developers.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: specs/global-plugin-system.yaml [hunk 1]
```diff
@@ -0,0 +1,615 @@
...
+        '500':
+          description: Internal Server Error, could not fetch the plugin details
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+
+
+
+  /orchestrator/plugin/global/:
```
Comment (gireesh-devtron) [inline specs/global-plugin-system.yaml:109]: check if we require ending slash

**Review by gireesh-devtron — DISMISSED** [review]

File: specs/global-plugin-system.yaml [hunk 2]
```diff
@@ -0,0 +1,615 @@
...
+          type: string
+          description: Optional. Specifies the stage of the CI/CD pipeline or both (CI/CD) where this plugin can be used.
+        pluginSteps:
+          type: array
+          items:
+            $ref: '#/components/schemas/PluginStepsDto'
+          description: Optional, A list of steps defined for the plugin. Each step contains specific instructions or actions the plugin will perform.
+      required:
+        - name
+        - description
```
[the remark this answers is not in the document]
Comment (RajeevRanjan27) [reply specs/global-plugin-system.yaml:277]: it is optional.

File: specs/global-plugin-system.yaml [hunk 3]
```diff
@@ -0,0 +1,615 @@
...
+          type: string
+          description: URL or a base64 encoded string representing an icon for the plugin
+          example: https://raw.githubusercontent.com/devtron-labs/devtron/main/assets/k6-plugin-icon.png
+        tags:
+          type: array
+          items:
+            type: string
+          description: A list of tags associated with the plugin for categorization and search.
+        action:
+          type: integer
```
[the remark this answers is not in the document]
Comment (RajeevRanjan27) [reply specs/global-plugin-system.yaml:255]: used by the users to group the similar plugin handling as per their convenience

File: specs/global-plugin-system.yaml [hunk 4]
```diff
@@ -0,0 +1,615 @@
...
+            - Use `1` to update an existing plugin.
+            - Use `2` to delete a plugin.
+    
+            If you're opting to update an existing plugin (action `1`), 
+            please take note:
+            - Ensure that the request body parameters are accurate before updating.
+            - Your request body (payload) will discard all previous updates and be treated as final.
+
+        pluginStage:
+          type: string
```
[the remark this answers is not in the document]
Comment (RajeevRanjan27) [reply specs/global-plugin-system.yaml:268]: added the enum

File: specs/global-plugin-system.yaml [hunk 5]
```diff
@@ -0,0 +1,617 @@
...
+          schema:
+            type: integer
+          description: Unique identifier of the plugin
+
+      responses:
+        '200':
+          description: successfully return the Detailed information about the plugin
+          content:
+            application/json:
+              schema:
```
Comment (RajeevRanjan27) [inline specs/global-plugin-system.yaml:27]: Status is not htere in theRESPONSE. HANDLE IT

**Pull request closed without merging**